Notify/Notify/Notify/ViewModels/BluetoothSettingsPageViewModel.cs

```diff
@@ -60,12 +99,82 @@ private async void onUpdateBluetoothSettingsClicked()
                         $"Updated {SelectedBluetoothID} as your {SelectedLocation}", "OK");
                     await AzureHttpClient.Instance.GetDestinations();
+                    reloadRemoveButton();
                 }
                 else
                 {
                     App.Current.MainPage.DisplayAlert("Error", "Something went wrong", "OK");
                 }
             }
         }
+        
+        #region Remove_Destination
+        
+        public Command RemoveBluetoothDestinationCommand { get; set; }
+        
+        private async void onRemoveBluetoothDestinationClicked()
+        {
+            bool isSucceeded;
+            bool isConfirmed = await App.Current.MainPage.DisplayAlert("Confirmation", $"Are you sure you want to remove the bluetooth device from your {SelectedLocation} destination?", "Yes", "No");
+
+            if (isConfirmed)
+            {
+                isSucceeded = AzureHttpClient.Instance
+                    .RemoveDestination(m_SelectedLocation, NotificationType.Bluetooth).Result;
+
+                if (isSucceeded)
+                {
+                    App.Current.MainPage.DisplayAlert("Remove Succeeded", $"Removal of bluetooth device from {SelectedLocation} succeeded", "OK");
+                    await AzureHttpClient.Instance.GetDestinations();
+                    reloadRemoveButton();
+                }
+                else
+                {
+                    App.Current.MainPage.DisplayAlert("Error", "Something went wrong", "OK");
+                }
+            }
+        }
+        
+        private string m_RemoveBluetoothButtonText = "CHOOSE DESTINATION";
+        public string RemoveBluetoothButtonText
+        {
+            get => m_RemoveBluetoothButtonText;
+            set => SetField(ref m_RemoveBluetoothButtonText, value);
+        }
+
+        private bool m_IsRemoveButtonEnabled;
+        public bool IsRemoveButtonEnabled
+        {
+            get => m_IsRemoveButtonEnabled;
+            set => SetField(ref m_IsRemoveButtonEnabled, value);
+        }
+        
+        #endregion
+        
+        #region Interface_Implementation
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        protected virtual void OnPropertyChanged([CallerMemberName] string propertyName = null)
+        {
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+        }
+
+        protected bool SetField<T>(ref T field, T value, [CallerMemberName] string propertyName = null)
+        {
+            if (EqualityComparer<T>.Default.Equals(field, value)) return false;
+            field = value;
+            OnPropertyChanged(propertyName);
+            return true;
+        }
+
+        #endregion
+        
+        private void reloadRemoveButton()
+        {
+            string currentDestination = SelectedLocation;
+            SelectedLocation = null;
+            SelectedLocation = currentDestination;
```

Comment:
I am not sure why you set the `SelectedLocation` to null and then immediately set its value to `currentDestination`